driver/examples/configs/baroclinic_c12_write_restart.yaml

```diff
@@ -0,0 +1,90 @@
+stencil_config:
+  backend: gtc:numpy
+  rebuild: false
+  validate_args: true
+  format_source: false
+  device_sync: true
+initialization:
+  type: baroclinic
+performance_config:
+  performance_mode: true
+  experiment_name: c12_baroclinic
+nx_tile: 12
+nz: 79
+dt_atmos: 225
+minutes: 0
+seconds: 225
+layout:
+  - 1
+  - 1
+save_restart: true
```

Comment:
Do we want/need to configure how often restarts are saved? It could slow down the simulation a lot to do this every timestep.

Reply:
Currently, it is called as part of driver cleanup, so only once. Because of the try/finally logic, I believe this would've saved the last time step before model crashes as well. 

I think for our use case, we just want to save the last time step. We could add logic to do restart frequency later if we have a use for that.

Reply:
Ah for now saving at the end should be good enough. There's decent reason to want to save only at the end of timesteps periodically, since an exception in the middle of a timestep could give an inconsistent model state, but we don't need to worry about that for now.